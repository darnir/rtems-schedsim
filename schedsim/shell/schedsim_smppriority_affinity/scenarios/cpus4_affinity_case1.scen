#
#  
#
rtems_init 4
task_create smp1 4 -a 0xe
task_create smp2 4 -a 0xe
task_create smp3 4 -a 0x3
task_create smp4 4 -a 0x3

# GOAL: want 0x0a010004 - 0x0a010001 as heir and executings
# GOAL:   on CPUs 0 -3 respectively
cpus

task_set_affinity smp1 0x03
# GOAL: want 0x0a010004, 0x0a010001, 0x0a010002,0x09010001
# GOAL:   as heir and executings on CPUs 0-3 respectively
cpus

task_set_affinity smp2 0x03
# GOAL: want 0x0a010004, 0x0a010002, 0x00910002,0x09010001
# GOAL:   as heir and executings on CPUs 0-3 respectively
cpus

task_set_affinity smp3 0x0e
# GOAL: want 0x0a010004, 0x0a010002, 0x09010002,0x0a010003
# GOAL:   as heir and executings on CPUs 0-3 respectively
cpus

task_set_affinity smp4 0x0e
# GOAL: want 0x0a010004, 0x0a010002, 0x0a010004,0x0a010003
# GOAL:   as heir and executings on CPUs 0-3 respectively
cpus
