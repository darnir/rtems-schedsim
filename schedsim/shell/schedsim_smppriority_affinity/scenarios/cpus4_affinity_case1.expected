Script File               : cpus4_affinity_case1.scen
verbose                   : 1
==> 1: #
==> 2: #  
==> 3: #
==> 4: rtems_init 4
  CPU 0: Thread Heir 0x09010001 priority=255
  CPU 0: Thread Executing 0x09010001 priority=255
  CPU 1: Thread Heir 0x09010002 priority=255
  CPU 1: Thread Executing 0x09010002 priority=255
  CPU 2: Thread Heir 0x09010003 priority=255
  CPU 2: Thread Executing 0x09010003 priority=255
  CPU 3: Thread Heir 0x09010004 priority=255
  CPU 3: Thread Executing 0x09010004 priority=255
==> 5: task_create smp1 4 -a 14
Creating task with:  Preemption: yes Timeslicing: no
Task (smp1) created: id=0x0a010001, priority=4
Task (0x0a010001) Set affinity=0x0000000e
Task (smp1) starting: id=0x0a010001, priority=4
  CPU 3: Thread Heir 0x0a010001 priority=4
  CPU 3: Thread Executing 0x0a010001 priority=4
==> 6: task_create smp2 4 -a 14
Creating task with:  Preemption: yes Timeslicing: no
Task (smp2) created: id=0x0a010002, priority=4
Task (0x0a010002) Set affinity=0x0000000e
Task (smp2) starting: id=0x0a010002, priority=4
  CPU 2: Thread Heir 0x0a010002 priority=4
  CPU 2: Thread Executing 0x0a010002 priority=4
==> 7: task_create smp3 4 -a  3
Creating task with:  Preemption: yes Timeslicing: no
Task (smp3) created: id=0x0a010003, priority=4
Task (0x0a010003) Set affinity=0x00000003
Task (smp3) starting: id=0x0a010003, priority=4
  CPU 1: Thread Heir 0x0a010003 priority=4
  CPU 1: Thread Executing 0x0a010003 priority=4
==> 8: task_create smp4 4 -a  3
Creating task with:  Preemption: yes Timeslicing: no
Task (smp4) created: id=0x0a010004, priority=4
Task (0x0a010004) Set affinity=0x00000003
Task (smp4) starting: id=0x0a010004, priority=4
  CPU 0: Thread Heir 0x0a010004 priority=4
  CPU 0: Thread Executing 0x0a010004 priority=4
==> 9: 
==> 10: # GOAL: want 0x0a010004 - 0x0a010001 as heir and executings
==> 11: # GOAL:   on CPUs 0 -3 respectively
==> 12: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010004 @  4 / 0x0a010004 @  4          false
  CPU 1: 0x0a010003 @  4 / 0x0a010003 @  4          false
  CPU 2: 0x0a010002 @  4 / 0x0a010002 @  4          false
  CPU 3: 0x0a010001 @  4 / 0x0a010001 @  4          false
=== End of Ready Set of Threads
==> 13: 
==> 14: task_set_affinity smp1 0x03
Task (0x0a010001) Set affinity=0x00000003
  CPU 1: Thread Heir 0x0a010001 priority=4
  CPU 1: Thread Executing 0x0a010001 priority=4
==> 15: # GOAL: want 0x0a010004, 0x0a010001, 0x0a010002,0x09010001
==> 16: # GOAL:   as heir and executings on CPUs 0-3 respectively
==> 17: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010004 @  4 / 0x0a010004 @  4          false
  CPU 1: 0x0a010001 @  4 / 0x0a010001 @  4          false
  CPU 2: 0x0a010002 @  4 / 0x0a010002 @  4          false
  CPU 3: 0x09010001 @  4 / 0x09010001 @  4          false
=== End of Ready Set of Threads
==> 18: 
==> 19: task_set_affinity smp2 0x03
Task (0x0a010002) Set affinity=0x00000003
  CPU 1: Thread Heir 0x0a010002 priority=4
  CPU 1: Thread Executing 0x0a010002 priority=4
==> 20: # GOAL: want 0x0a010004, 0x0a010002, 0x00910002,0x09010001
==> 21: # GOAL:   as heir and executings on CPUs 0-3 respectively
==> 22: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010004 @  4 / 0x0a010004 @  4          false
  CPU 1: 0x0a010002 @  4 / 0x0a010002 @  4          false
  CPU 2: 0x09010002 @  4 / 0x09010002 @  4          false
  CPU 3: 0x09010001 @  4 / 0x09010001 @  4          false
=== End of Ready Set of Threads
==> 23: 
==> 24: task_set_affinity smp3 0x0e
Task (0x0a010003) Set affinity=0x0000000e
  CPU 3: Thread Heir 0x0a010003 priority=4
  CPU 3: Thread Executing 0x0a010003 priority=4
==> 25: # GOAL: want 0x0a010004, 0x0a010002, 0x09010002,0x0a010003
==> 26: # GOAL:   as heir and executings on CPUs 0-3 respectively
==> 27: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010004 @  4 / 0x0a010004 @  4          false
  CPU 1: 0x0a010002 @  4 / 0x0a010002 @  4          false
  CPU 2: 0x09010002 @  4 / 0x09010002 @  4          false
  CPU 3: 0x0a010003 @  4 / 0x0a010003 @  4          false
=== End of Ready Set of Threads
==> 28: 
==> 29: task_set_affinity smp4 0x0e
Task (0x0a010004) Set affinity=0x0000000e
  CPU 2: Thread Heir 0x0a010004 priority=4
  CPU 2: Thread Executing 0x0a010004 priority=4
==> 30: # GOAL: want 0x0a010004, 0x0a010002, 0x0a010004,0x0a010003
==> 31: # GOAL:   as heir and executings on CPUs 0-3 respectively
==> 32: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010004 @  4 / 0x0a010004 @  4          false
  CPU 1: 0x0a010002 @  4 / 0x0a010002 @  4          false
  CPU 2: 0x0a010004 @  4 / 0x0a010004 @  4          false
  CPU 3: 0x0a010003 @  4 / 0x0a010003 @  4          false
=== End of Ready Set of Threads
